### ISSUE TITLE
Fix issue when compiling with GCC 12

### ISSUE THREAD

**Issue by tuliom** [libnxz/power-gzip#137]
Issue found on Fedora Rawhide

```
nx_utils.c: In function 'find_cfg':
nx_utils.c:102:21: error: the comparison will always evaluate as 'true' for the address of 'key' will never be NULL [-Werror=address]
  102 |                 if (!configs[i].key)
      |                     ^
nx_utils.c:49:14: note: 'key' declared here
   49 |         char key[MAX_KEY_LEN];
      |              ^~~
nx_utils.c: In function 'set_cfg':
nx_utils.c:141:13: error: the comparison will always evaluate as 'true' for the address of 'key' will never be NULL [-Werror=address]
  141 |         if (!configs[cfg_cnt].key || !configs[cfg_cnt].val)
      |             ^
nx_utils.c:49:14: note: 'key' declared here
   49 |         char key[MAX_KEY_LEN];
      |              ^~~
nx_utils.c:141:38: error: the comparison will always evaluate as 'true' for the address of 'val' will never be NULL [-Werror=address]
  141 |         if (!configs[cfg_cnt].key || !configs[cfg_cnt].val)
      |                                      ^
nx_utils.c:50:14: note: 'val' declared here
   50 |         char val[MAX_CONFIG_LINE];
      |              ^~~
nx_utils.c: In function 'nx_dump_cfg':
nx_utils.c:163:24: error: the comparison will always evaluate as 'true' for the address of 'key' will never be NULL [-Werror=address]
  163 |                 key = (!cfg_table->configs[i].key) ? "NULL" : cfg_table->configs[i].key;
      |                        ^
nx_utils.c:49:14: note: 'key' declared here
   49 |         char key[MAX_KEY_LEN];
      |              ^~~
nx_utils.c:164:24: error: the comparison will always evaluate as 'true' for the address of 'val' will never be NULL [-Werror=address]
  164 |                 val = (!cfg_table->configs[i].val) ? "NULL" : cfg_table->configs[i].val;
      |                        ^
nx_utils.c:50:14: note: 'val' declared here
   50 |         char val[MAX_CONFIG_LINE];
      |              ^~~
```

Source: https://bugzilla.redhat.com/show_bug.cgi?id=2045820

**Comment by mscastanho** [comment]
@tuliom I suspect these have already been fixed in the `develop` branch on commit 28c9671c39d50934da68c16a8c6730288e53e11e

**Comment by tuliom** [comment]
@mscastanho this patch does solve the issue. Thanks!
